crates/pera-core/src/dwallet_mpc/dkg.rs

```diff
@@ -48,27 +54,57 @@ impl FirstDKGBytesParty {
     }
 }
 
+/// Deserializes the messages received from other parties for the next advancement.
+/// Any value that fails to deserialize is considered to be sent by a malicious party.
+/// Returns the deserialized messages or an error including the IDs of the malicious parties.
+pub fn deserialize_mpc_messages<M: DeserializeOwned + Clone>(
+    messages: HashMap<PartyID, Vec<u8>>,
+) -> PeraResult<HashMap<PartyID, M>> {
+    let parsing_results: Vec<PeraResult<(PartyID, _)>> = messages
+        .iter()
+        .map(|(k, v)| {
+            let value =
+                bcs::from_bytes(&v).map_err(|_| PeraError::DWalletMPCMaliciousParties(vec![*k]))?;
+            Ok((*k, value))
+        })
+        .collect();
+    let malicious_parties: Vec<PartyID> = parsing_results
+        .clone()
+        .into_iter()
+        .filter_map(|result| {
+            if let Err(PeraError::DWalletMPCMaliciousParties(malicious_parties)) = result {
+                Some(malicious_parties)
+            } else {
+                None
+            }
+        })
+        .flatten()
+        .collect();
+    if !malicious_parties.is_empty() {
+        return Err(PeraError::DWalletMPCMaliciousParties(malicious_parties));
+    }
+    parsing_results.into_iter().collect()
```

Comment:
```suggestion
    if malicious_parties.is_empty() {
        parsing_results.into_iter().collect()
    } else {
        Err(PeraError::DWalletMPCMaliciousParties(malicious_parties))
    }

```

Reply:
Zeev prefer to return early instead of using if else